tgg/seqr_scripts/sex_check.py

```diff
@@ -0,0 +1,334 @@
+
+    mt = hl.read_matrix_table(callset)
+
+    # Filter to SNVs and biallelics
+    mt = mt.filter_rows(
+        (hl.len(mt.alleles) == 2) & hl.is_snp(mt.alleles[0], mt.alleles[1])
+    )
+
+    # Filter to pass variants only (empty set)
+    # TODO: Make this an optional argument before moving to gnomad_methods
+    mt = mt.filter_rows(mt.filters.length() == 0, keep=True)
+
+    # Infer build:
+    build = get_reference_genome(mt.locus).name
+    logger.info("Build inferred as %s", build)
+
+    logger.info("Inferring sex...")
+    if use_y_cov:
+        norm_ht = get_chr_cov(mt, "GRCh38", normalization_contig, call_rate_threshold)
+        mt = mt.annotate_cols(**norm_ht[mt.col_key])
+        chry_ht = get_chr_cov(mt, "GRCh38", "Y", call_rate_threshold)
+        mt = mt.annotate_cols(**chry_ht[mt.col_key])
+        mt = mt.annotate_cols(
+            normalized_y_coverage=hl.or_missing(
+                mt[f"chr{normalization_contig}_mean_dp"] > 0,
+                mt.chrY_mean_dp / mt[f"chr{normalization_contig}_mean_dp"],
+            )
+        )
+        if add_x_cov:
+            chrx_ht = get_chr_cov(mt, "GRCh38", "X", call_rate_threshold)
+            mt = mt.annotate_cols(**chrx_ht[mt.col_key])
+            mt = mt.annotate_cols(
+                normalized_x_coverage=hl.or_missing(
+                    mt[f"chr{normalization_contig}_mean_dp"] > 0,
+                    mt.chrX_mean_dp / mt[f"chr{normalization_contig}_mean_dp"],
+                )
+            )
+        sex_ht = run_hails_impute_sex(
+            mt,
+            build,
+            outdir,
+            mt_name,
+            xy_fstat_threshold,
+            xx_fstat_threshold,
+            aaf_threshold,
+        )
+        sex_ht = sex_ht.annotate(
+            ambiguous_sex=hl.is_missing(sex_ht.is_female),
+            sex_aneuploidy=(sex_ht.is_female)
+            & hl.is_defined(sex_ht.normalized_y_coverage)
+            & (sex_ht.normalized_y_coverage > y_cov_threshold)
+            | (~sex_ht.is_female)
+            & hl.is_defined(sex_ht.normalized_y_coverage)
+            & (sex_ht.normalized_y_coverage < y_cov_threshold),
+        )
+
+        sex_expr = (
+            hl.case()
+            .when(sex_ht.ambiguous_sex, "ambiguous_sex")
+            .when(sex_ht.sex_aneuploidy, "sex_aneuploidy")
+            .when(sex_ht.is_female, "XX")
+            .default("XY")
+        )
+        sex_ht = sex_ht.annotate(sex=sex_expr)
+
+    else:
+        sex_ht = run_hails_impute_sex(
+            mt,
+            build,
+            outdir,
+            mt_name,
+            xy_fstat_threshold,
+            xx_fstat_threshold,
+            aaf_threshold,
+        )
+        sex_ht = sex_ht.annotate(ambiguous_sex=hl.is_missing(sex_ht.is_female))
+        sex_expr = hl.if_else(
+            sex_ht.ambiguous_sex,
+            "ambiguous_sex",
+            hl.if_else(sex_ht.is_female, "XX", "XY"),
+        )
+        sex_ht = sex_ht.annotate(sex=sex_expr)
```

Comment:
```suggestion
    sex_ht = sex_ht.annotate(sex=sex_expr)
```
unindenting this (and deleting the line above) because this line gets run both if using y coverage and if not using y coverage, and this way, we only need to have this line in the code once